### PR TITLE
Changes to the TwimCal2Hit and TwinHitData classes to save the offset…

### PR DESCRIPTION
**Pull request by antia1996**
… parameter. Added new class twim/ana/TwimVertexReconstruction to find the fission position

### CODE REVIEW

File: r3bdata/twimData/R3BTwimHitData.cxx [hunk 1]
```diff
@@ -24,17 +24,21 @@ R3BTwimHitData::R3BTwimHitData()
...
 // -----   Standard constructor   ------------------------------------------
-R3BTwimHitData::R3BTwimHitData(UInt_t secID, Double_t theta, Double_t z, Double_t x, Double_t ene)
+R3BTwimHitData::R3BTwimHitData(UInt_t secID, Double_t theta, Double_t z, Double_t x, Double_t ene, Double_t offset, Double_t vertex)
     : fSecID(secID)
     , fTheta(theta)
     , fZ(z)
     , fE(ene)
     , fX(x)
+    , fOffset(offset)
+    , fVertex(0.)
```
Comment (ryotani) [inline r3bdata/twimData/R3BTwimHitData.cxx:41]: `, fVertex(vertex)` ?

File: twim/calibration/R3BTwimCal2Hit.cxx [hunk 2]
```diff
@@ -811,25 +800,30 @@ void R3BTwimCal2Hit::Reset()
...
+                                           Double_t charge_z,
+                                           Double_t xpos,
+                                           Double_t offset)
 {
     // It fills the R3BTwimHitData
     TClonesArray& clref = *fTwimHitDataCA;
     Int_t size = clref.GetEntriesFast();
-    return new (clref[size]) R3BTwimHitData(secid, theta, charge_z, xpos);
+    return new (clref[size]) R3BTwimHitData(secid, theta, charge_z, xpos, offset);
 }
```
Comment (ryotani) [inline twim/calibration/R3BTwimCal2Hit.cxx:813]: I think we can remove this as this function is anymore used in this code?

File: twim/calibration/R3BTwimCal2Hit.h [hunk 3]
```diff
@@ -110,8 +110,13 @@ class R3BTwimCal2Hit : public FairTask
 
     /** Private method TwimHitData **/
     //** Adds a TwimHitData to the detector
-    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos);
-    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos, Double_t ene_ave);
+    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos, Double_t offset);
```
Comment (ryotani) [inline twim/calibration/R3BTwimCal2Hit.h:113]: If this is not used, shall we delete this line and merge it into the next function?

File: twim/calibration/R3BTwimCal2Hit.h [hunk 4]
```diff
@@ -110,8 +110,13 @@ class R3BTwimCal2Hit : public FairTask
...
+    R3BTwimHitData* AddHitData(UShort_t secID,
+                               Double_t theta,
+                               Double_t charge_z,
+                               Double_t xpos,
+                               Double_t ene_ave,
+                               Double_t offset);
```
Comment (ryotani) [inline twim/calibration/R3BTwimCal2Hit.h:120]: Then add default values for the last two arguments?
`    R3BTwimHitData* AddHitData(UShort_t secID,
                               Double_t theta,
                               Double_t charge_z,
                               Double_t xpos,
                               Double_t ene_ave = 0.0,
                               Double_t offset = 0.0);`

**Review by ryotani — COMMENTED** [review]
Just some suggestions for improvement.

**Comment by antia1996** [comment]
Hi Ryo,

Yes sure, thank you.

Antía
________________________________
De: Ryo Taniuchi ***@***.***>
Enviado: miércoles, 27 de julio de 2022 21:55
Para: R3BRootGroup/R3BRoot ***@***.***>
Cc: GRAÑA GONZALEZ ANTIA ***@***.***>; Author ***@***.***>
Asunto: Re: [R3BRootGroup/R3BRoot] Changes to the TwimCal2Hit and TwinHitData classes to save the offset… (PR #644)


@ryotani commented on this pull request.

Just some suggestions for improvement.

________________________________

In r3bdata/twimData/R3BTwimHitData.cxx<https://github.com/R3BRootGroup/R3BRoot/pull/644#discussion_r931521640>:

>      : fSecID(secID)
     , fTheta(theta)
     , fZ(z)
     , fE(ene)
     , fX(x)
+    , fOffset(offset)
+    , fVertex(0.)


, fVertex(vertex) ?

________________________________

In twim/calibration/R3BTwimCal2Hit.cxx<https://github.com/R3BRootGroup/R3BRoot/pull/644#discussion_r931526667>:

> +R3BTwimHitData* R3BTwimCal2Hit::AddHitData(UShort_t secid,
+                                           Double_t theta,
+                                           Double_t charge_z,
+                                           Double_t xpos,
+                                           Double_t offset)
 {
     // It fills the R3BTwimHitData
     TClonesArray& clref = *fTwimHitDataCA;
     Int_t size = clref.GetEntriesFast();
-    return new (clref[size]) R3BTwimHitData(secid, theta, charge_z, xpos);
+    return new (clref[size]) R3BTwimHitData(secid, theta, charge_z, xpos, offset);
 }


I think we can remove this as this function is anymore used in this code?

________________________________

In twim/calibration/R3BTwimCal2Hit.h<https://github.com/R3BRootGroup/R3BRoot/pull/644#discussion_r931527005>:

> @@ -110,8 +110,13 @@ class R3BTwimCal2Hit : public FairTask

     /** Private method TwimHitData **/
     //** Adds a TwimHitData to the detector
-    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos);
-    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos, Double_t ene_ave);
+    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos, Double_t offset);


If this is not used, shall we delete this line and merge it into the next function?

________________________________

In twim/calibration/R3BTwimCal2Hit.h<https://github.com/R3BRootGroup/R3BRoot/pull/644#discussion_r931527461>:

> @@ -110,8 +110,13 @@ class R3BTwimCal2Hit : public FairTask

     /** Private method TwimHitData **/
     //** Adds a TwimHitData to the detector
-    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos);
-    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos, Double_t ene_ave);
+    R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos, Double_t offset);
+    R3BTwimHitData* AddHitData(UShort_t secID,
+                               Double_t theta,
+                               Double_t charge_z,
+                               Double_t xpos,
+                               Double_t ene_ave,
+                               Double_t offset);



Then add default values for the last two arguments?
R3BTwimHitData* AddHitData(UShort_t secID, Double_t theta, Double_t charge_z, Double_t xpos, Double_t ene_ave = 0.0, Double_t offset = 0.0);

**Pull request merged**